src/reducers/contacts.js

```diff
@@ -24,9 +30,30 @@ export default (state: State = initialState, action: Action): State => {
   switch (action.type) {
     case SERVICES_DESTROYED:
       return initialState;
-      // @todo Implement action handling
+    case START_CONTACTS_FETCH:
+      return {
+        ...state,
+        isFetching: true,
+        fetchError: null,
+      };
+    case CONTACTS_FETCH_FAILED:
+      return {
+        ...state,
+        isFetching: false,
+        fetchError: action.error,
+      };
+    case CONTACTS_UPDATED:
+      return {
+        ...state,
+        isFetching: false,
+        contacts: [...action.contacts],
+      };
+    case ADD_CONTACT:
+      return {
+        ...state,
+        isFetching: false,
```

Comment:
I'm not sure why you reset isFetching flag if you add contact.

Reply:
@seland  Removed it